File: rpc/java/evoxbench/src/main/java/evoxgroup/IN1KMOP.java
package evoxgroup;
import evoxgroup.EvoXException;
import evoxgroup.Benchmark;
import evoxgroup.BenchmarkFactory;
import evoxgroup.Config;
import evoxgroup.Args;

public class IN1KMOP extends BenchmarkFactory {
    IN1KMOP(String address, int port) {
        super(address, port);
    }

    Benchmark createBenchmark(int problemId) throws EvoXException {
        Config config;
        Args args = new Args();

        switch (problemId) {
            case 1:
                args.objs = "err&params";
                args.normalized_objectives = false;
                config = new Config("resnet", args);
                break;
            case 2:
                Args args = new Args();
                args.objs = "err&flops";
                args.normalized_objectives = false;
                config = new Config("resnet", args);
                break;
            case 3:
                Args args = new Args();
                args.objs = "err&params&flops";
                args.normalized_objectives = false;
                config = new Config("resnet", args);
                break;
            case 7:
                Args args = new Args();
                args.objs = "err&params";
                args.normalized_objectives = false;
                config = new Config("mnv3", args);
                break;
            case 8:
                Args args = new Args();
                args.objs = "err&params&flops";
                args.normalized_objectives = false;
                config = new Config("mnv3", args);
                break;
            case 9:
                Args args = new Args();
                args.objs = "err&params&flops&latency";
                args.normalized_objectives = false;
                config = new Config("mnv3", args);
                break;
            default:
                throw new EvoXException("No such problem id");
        }
        Benchmark bm = new Benchmark();
        bm.init(address, port, config);

        return bm;
    }
}

File: evoxbench/test_suites/in1kmop.py
from evoxbench.benchmarks import ResNet50DBenchmark, MobileNetV3Benchmark, TransformerBenchmark

__all__ = ['in1kmop']


def in1kmop(problem_id):
    if problem_id == 1:
        return ResNet50DBenchmark(
            objs='err&params', normalized_objectives=False)
    elif problem_id == 2:
        return ResNet50DBenchmark(
            objs='err&flops', normalized_objectives=False)
    elif problem_id == 3:
        return ResNet50DBenchmark(
            objs='err&params&flops', normalized_objectives=False)
    elif problem_id == 4:
        return TransformerBenchmark(       
            objs='err&params', normalized_objectives=False)
    elif problem_id == 5:
        return TransformerBenchmark(
            objs='err&params', normalized_objectives=False)
    elif problem_id == 6:
        return TransformerBenchmark(
            objs='err&params', normalized_objectives=False)
    elif problem_id == 7:
        return MobileNetV3Benchmark(
            objs='err&params', normalized_objectives=False)
    elif problem_id == 8:
        return MobileNetV3Benchmark(
            objs='err&params&flops', normalized_objectives=False)
    elif problem_id == 9:
        return MobileNetV3Benchmark(
            objs='err&params&flops&latency', normalized_objectives=False)
    else:
        raise ValueError("the requested problem id does not exist")
